### ISSUE TITLE
New Feature: Analyze Game

### ISSUE THREAD

**Issue by MarkZH** [MarkZH/Genetic_Chess#13]
Pick a game to replay and have the players write out their analysis of each move to a new file. Analysis should include the predicted final board state for each move and the scores for each.

`genetic_chess --analyze_game <game_file>.txt <game_id>`

<game_id> can consist of a game number (not implemented yet) or the IDs of the white and black player in order.

**Comment by MarkZH** [comment]
More interesting output: for each move, the player making the move should also print out how it expected the subsequent moves to go based on the most favorable board outcome.

```
1. e4 ... {1. ... e5 2. Nf3 Nc6 }
```
This should also give an idea of how far ahead the players are looking.

Implementation:

```
class Genetic_AI : public Player
{
    // ...
    private:
        static ofstream Genetic_AI::thinking_output;
        std::vector<std::string> most_likely_line;
    // ...
};
```

```
const Complete_Move Genetic_AI::choose_move(const Board& board, const Clock& clock) const
{
    most_likely_line.clear()
    auto legal_moves = board.all_legal_moves();
    if(legal_moves.size() == 1)
    {
        thinking_output << legal_moves.front().game_record_item() << "{ only legal move }\n";
        return legal_moves.front(); // If there's only one legal move, take it.
    }

    auto returned_move = std::get<Complete_Move>(search_game_tree(board,
                                                                  genome.positions_to_examine(board, clock),
                                                                  clock));
    thinking_output << move.game_record_item() << " { ";
    for(const auto& move : most_likely_line)
    {
        thinking_output << move.game_record_item() << " ";
    }
    thinking_output << "}\n";
    return returned_move;
}
```
The above needs more formatting.

**Comment by MarkZH** [comment]
Implemented in commit 1f6d90980ee9a95576ff05741a239c579cada4b8.

**Comment by MarkZH** [comment]
Completed implementation with commit 8bb6beeb7e9b679275141ab6322ecf1e3e89e60e.
